app.go

```diff
@@ -1302,3 +1315,101 @@ func (app *App) printRoutesMessage() {
 
 	_ = w.Flush()
 }
+
+// OnRoute is a hook to execute user functions on each route registeration.
+// Also you can get route properties by "route" key of map.
+func (app *App) OnRoute(handler ...HookHandler) {
+	app.mutex.Lock()
+	app.hookList["onRoute"] = append(app.hookList["onRoute"], handler...)
+	app.mutex.Unlock()
+}
+
+// OnName is a hook to execute user functions on each route naming.
+// Also you can get route properties by "route" key of map.
+//
+// WARN: OnName only works with naming routes, not groups.
+func (app *App) OnName(handler ...HookHandler) {
+	app.mutex.Lock()
+	app.hookList["onName"] = append(app.hookList["onName"], handler...)
+	app.mutex.Unlock()
+}
+
+// OnListen is a hook to execute user functions on Listen, ListenTLS, Listener.
+func (app *App) OnListen(handler ...HookHandler) {
+	app.mutex.Lock()
+	app.hookList["onListen"] = append(app.hookList["onListen"], handler...)
+	app.mutex.Unlock()
+}
+
+// OnShutdown is a hook to execute user functions after Shutdown.
+func (app *App) OnShutdown(handler ...HookHandler) {
+	app.mutex.Lock()
+	app.hookList["onShutdown"] = append(app.hookList["onShutdown"], handler...)
+	app.mutex.Unlock()
+}
+
+// OnResponse is a hook to execute user functions after a response.
+//
+// WARN: You can't edit response with OnResponse hook.
+func (app *App) OnResponse(handler ...HookHandler) {
+	app.mutex.Lock()
+	app.hookList["onResponse"] = append(app.hookList["onResponse"], handler...)
+	app.mutex.Unlock()
+}
+
+// OnRequest is a hook to execute user functions after a request.
+//
+// WARN: You can edit response with OnRequest hook.
+func (app *App) OnRequest(handler ...HookHandler) {
+	app.mutex.Lock()
+	app.hookList["onRequest"] = append(app.hookList["onRequest"], handler...)
+	app.mutex.Unlock()
+}
+
+func (app *App) executeOnRouteHooks(route Route) {
+	for _, v := range app.hookList["onRoute"] {
+		ctx := app.AcquireCtx(&fasthttp.RequestCtx{})
+		defer app.ReleaseCtx(ctx)
+
+		v(ctx, Map{"route": route})
```

Comment:
I'm not sure about what to do here. Should i skip error handling or do anything else?